src/main/java/tds/exam/results/mappers/TestMapper.java

```diff
@@ -38,4 +51,53 @@ private static String parseBankKeyFromId(final String itemId) {
 
         return itemId.split("-")[0];
     }
+
+    private static long parseLatestAcademicYear(final String academicYearString) {
+        // If this is a range date, pick the latest year
+        if (academicYearString.contains("-")) {
+            return Long.parseLong(academicYearString.split("-")[1]);
+        }
+
+        return Long.parseLong(academicYearString);
+    }
+
+    private static String createGradeStringFromGrades(final List<String> grades) {
+        if (grades.size() == 0) {
```

Comment:
Forgot to add this as a comment in the code, but this logic is ported over from ReportingDLL.TestGradeSpan_F(), line 1353